MonMaraicher/Features/Map/MapView.swift

```diff
@@ -13,22 +13,24 @@ struct MapView: View {
     @StateObject private var viewModel = MapViewModel(farmerService: FarmerService())
 
     var body: some View {
-        Map(position: $viewModel.mapCameraPosition, selection: $viewModel.selectedFarmer) {
+        Map(position: $viewModel.mapCameraPosition, selection: $viewModel.selectedAddress) {
 
             ForEach(viewModel.allFarmers, id: \.id) { farmer in
-                Marker(farmer.title,
-                       systemImage: farmer.systemImageName,
-                       coordinate: CLLocationCoordinate2D(latitude: farmer.location.latitude, longitude: farmer.location.longitude))
-                .tag(farmer)
-                .tint(.orange)
+                ForEach(farmer.adressesOperateurs, id: \.id) { location in
+                    Marker(farmer.raisonSociale.capitalized, systemImage: "laurel.leading", coordinate: .init(latitude: location.lat, longitude: location.long))
+                        .tag(location)
+                        .tint(.orange)
+                }
             }
 
             UserAnnotation()
 
         }
-        .sheet(item: $viewModel.selectedFarmer) { farmer in
-            FarmerDetailsView(viewModel: FarmerDetailsViewModel(farmer: farmer))
-                .presentationDetents([.medium, .large])
+        .sheet(item: $viewModel.selectedAddress) { location in
+            if let selectedFarmer = viewModel.allFarmers.first(where: { $0.adressesOperateurs.contains(location) }) {
```

Comment:
That logic should go in your viewModel

Reply:
Your viewModel could create the FarmerDetailsViewModel for you. Instead of listening `$viewModel.selectedAddress`, you could have 
`@Published var farmerDetailsViewModel: FarmerDetailsViewModel?`
and
`.sheet(item: $viewModel.farmerDetailsViewModel)` { ... }